python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -970,6 +993,42 @@ cdef class Column:
 
         flat, shape = _flatten_nested_list(obj, depth)
 
+        if dtype.id() == type_id.STRING:
+            encoded = [s.encode("utf-8") for s in flat]
+            offsets = [0] + list(accumulate(len(s) for s in encoded))
+            joined_bytes = b"".join(encoded)
+
+            offsets_col = Column.from_iterable_of_py(
+                offsets, dtype=DataType(type_id.INT32)
```

Comment:
Not supporting large strings here (int64 offsets)? Perhaps iterable strings cannot be larger than max int32?

Reply:
There's not a hard limit on the python side.

Reply:
Let's punt this to a follow-up PR. Because it requires some additional redesign in the Column layer. I'll open an issue to track.

Reply:
I think it is just a matter of using `type_id.INT64` here if the last offset value `offsets[-1]` is greater than max int32.

Reply:
Thanks for the clarification. I'll try implementing this and follow back up with you

Reply:
Done in https://github.com/rapidsai/cudf/pull/18916/commits/bb54d973ef58a4bdcd82ad5763486307c06107ae